### DISCUSSION TITLE
lupa password

### DISCUSSION THREAD

**Discussion by azzamy-bendungan** [nafiesl/silsilah#91] [General]
gimana cara mengatasi lupa password ? klik lupa password, email tidak terkirim

**Comment by nafiesl** [comment]
### Penyebab fitur Email tidak aktif

Halo mas @azzamy-cell. Untuk fitur lupa password, kita harus melakukan konfigurasi email terlebih dahulu. Karena secara default konfigurasi email di project silsilah ini tidak aktif. Lihat pada file `.env` pada variable berikut ini:
```
MAIL_DRIVER=smtp
MAIL_HOST=smtp.mailtrap.io
MAIL_PORT=2525
MAIL_USERNAME=null
MAIL_PASSWORD=null
MAIL_ENCRYPTION=null
```

Untuk mengaktifkan fitur email mas bisa meng-update variable-variabel tersebut sesuai dengan penyedia layanan email yang mas gunakan.

### Saran Solusi

Untuk solusi yang umum digunakan, bisa menggunakan SMTP dari gmail. Atau jika mau menggunakan hosting dan domain sendiri, bisa melihat konfigurasi SMTP email dari cpanelnya.

Referensi konfigurasi gmail:
- https://codelapan.com/post/how-to-send-email-with-gmail-in-laravel
- https://codingspoint.com/how-to-set-gmail-configuration-for-mail-in-laravel

**Reply by azzamy-bendungan** [reply]
kalau mau melihat password dari phpmyadmin bisa gak?

**Reply by nafiesl** [reply]
@azzamy-cell tidak bisa mas, karena passwordnya disimpan dalam bentuk hash (seperti gambar di bawah).

Jika ingin melakukan reset password, mas bisa copas text ini ke kolom `password` di table `users`. Sesuai password yang ingin diubah:
```
$2y$10$De5xqwdx2pKNms821/sGSuy/8BbU5DJvlvtm7AQY7vlj8PjkPYLmm
```
Kemudian login dengan email yang dan passwordnya: `password123` (gambar terlampir di bawah).

Gambar kolom passowrd pada tabel users

![screen_2023-08-19_005](https://github.com/nafiesl/silsilah/assets/8721551/a9d8296b-6f04-42b5-8ca1-3945d29c1fad)

Gambar form login dengan password baru 

![screen_2023-08-19_004](https://github.com/nafiesl/silsilah/assets/8721551/fc204f75-d004-43c7-a36d-ab36ea2161c6)

Mohon maaf saya baru lihat pesannya ini mas @azzamy-cell, mudah2an membantu.

**Comment by SilsilahKerabat** [comment]
Berminat sekali dengan aplikasi silsilah...tapi nggak ngerti sama sekali..

**Reply by nafiesl** [reply]
Halo pak @SilsilahKerabat, software silsilah ini berbasis web, saran saya bapak bisa coba software ini di laptop atau PC dulu. Jika bapak menggunakan Windows, bisa coba mengikuti artikel ini untuk proses installnya:

https://blog.nafies.id/laravel/cara-install-aplikasi-laravel-dari-github-ke-xampp-windows/

Terima kasih.
